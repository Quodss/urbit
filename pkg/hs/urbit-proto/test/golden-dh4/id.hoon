=/  id
  |=  {t=$ x=t}
  x
:*  (id @ 1)
    (id * 2)
    (id {@ @} [3 4])
    (id ^ [5 6])
    (id ? &)
    (id $-({t=$ t} t) id)
    ((id $-({t=$ t} t) id) $-({t=$ t} t) id)
    (id $-({t=$ t} t) (id $-({t=$ t} t) id))
==
