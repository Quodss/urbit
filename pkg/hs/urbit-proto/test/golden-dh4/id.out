PARSED

=/  id
  |=  {t=$ x=t}
  ^-  t
  x
:*  (id @ 1)
    (id * 2)
    (id {@ @} [3 4])
    (id ^ [5 6])
    (id ? %&)
    (id $-({t=$ t} t) id)
    ((id $-({t=$ t} t) id) $-({t=$ t} t) id)
    (id $-({t=$ t} t) (id $-({t=$ t} t) id))
==

SOFT

=+  ^=  id
    |=  {t=$ x=t}
    ^-  t
    x
:*  (id @ 1)
    (id * 2)
    (id {@ @} 3 4)
    (id [* *] 5 6)
    (id ? %&)
    (id $-({t=$ t} t) id)
    ((id $-({t=$ t} t) id) $-({t=$ t} t) id)
    (id $-({t=$ t} t) (id $-({t=$ t} t) id))
==

HARD

=+  ^=  id
    =:  {t=$ x=t}
      ^-  *
      0
    |%
    ++  %
      ^-  t
      x
    --
:*  =+  id
    %.+2(+6 [@ 1]:+3)
    =+  id
    %.+2(+6 [* 2]:+3)
    =+  id
    %.+2(+6 [{@ @} 3 4]:+3)
    =+  id
    %.+2(+6 [[* *] 5 6]:+3)
    =+  id
    %.+2(+6 [? %&]:+3)
    =+  id
    %=  %.+2
        +6
      =>  +3
      :-  $?  +7
          $|  [$;({t=$ t} *) *]
          ++  %  t
          --
      id
    ::
    ==
    =+  =+  id
        %=  %.+2
            +6
          =>  +3
          :-  $?  +7
              $|  [$;({t=$ t} *) *]
              ++  %  t
              --
          id
        ::
        ==
    %=  %.+2
        +6
      =>  +3
      :-  $?  +7
          $|  [$;({t=$ t} *) *]
          ++  %  t
          --
      id
    ::
    ==
    =+  id
    %=  %.+2
        +6
      =>  +3
      :-  $?  +7
          $|  [$;({t=$ t} *) *]
          ++  %  t
          --
      =+  id
      %=  %.+2
          +6
        =>  +3
        :-  $?  +7
            $|  [$;({t=$ t} *) *]
            ++  %  t
            --
        id
      ::
      ==
    ::
    ==
==

CODE

=+  |=(_ +13)
:*  (+2 @ 1)
    (+2 * 2)
    (+2 {@ @} 3 4)
    (+2 [* *] 5 6)
    (+2 ? 0)
    (+2 $-({t=$ +2} +8) +2)
    %+  (+2 $-({t=$ +2} +8) +2)
      $-({t=$ +2} +8)
    +2
    %+  +2
      $-({t=$ +2} +8)
    (+2 $-({t=$ +2} +8) +2)
==

TYPE

:*  @
    *
    {@ @}
    [* *]
    ?
    $-({t=$ +2} +8)
    $-({t=$ +2} +8)
    $-({t=$ +2} +8)
==

BASE

:*  1
    2
    [3 4]
    [5 6]
    0
    |%
    ++  %  +13
    --
    |%
    ++  %  +13
    --
    |%
    ++  %  +13
    --
==